Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/model/Version.java

```diff
@@ -37,6 +37,40 @@ public int compareTo(Version that) {
         return 0;
     }
 
+    public boolean compatibleWith(Version that) {
+        if (that == null)
+            return false;
+
+        // this version is compatible with that version if:
+            // this.major = that.major
+            // this.minor >= that.minor
+        String[] thisParts = this.version.split("\\.");
+        String[] thatParts = that.version.split("\\.");
+        int length = Math.max(thisParts.length, thatParts.length);
+        for (int i = 0; i < length; i++) {
+            int thisPart = i < thisParts.length ? Integer.parseInt(thisParts[i]) : 0;
+            int thatPart = i < thatParts.length ? Integer.parseInt(thatParts[i]) : 0;
+            switch (i) {
+                case 0: {
+                    // Major version
+                    if (thisPart != thatPart)
+                        return false;
+                }
```

Comment:
Note that Java switch statements fall through until they hit a `return` or a `break`.  So, if the major case (`0`) is hit, but the parts are equal, it will fall through and execute the code in the minor case (`1`).  Now... since it's impossible for a part to be `==` and `>` at the same time, and since the patch case (`2`) is empty, this logic actually works as-written, but...  it _looks_ like a bug, and I think that in order to show the intent, there really should be a break at the end of `case 0` and `case 1`.

Reply:
Totally agree, great catch. I'm too used to C# where the compiler will tell me when I've missed those.